src/packet/fragment_buffer.rs

```diff
@@ -0,0 +1,77 @@
+use std::io::Result;
+use std::clone::Clone;
+
+/// Collection for storing data fof any kind.
+pub struct FragmentBuffer<T>  where T: Default + Clone + Send + Sync  {
+    entries: Vec<T>,
+    entry_sequences: Vec<u32>,
+    size: usize,
```

Comment:
Again not quite sure but will this `usize` cause us problems down the line? Just a thought.

Reply:
No don't think so Vec uses it also to have an capacity given by usize value. This represents the size of the buffer we have set with with_capacity previously.